controllers/topics/getLatestPost.js

```diff
@@ -64,12 +66,32 @@ module.exports = async (req, res) => {
 
     let message = `There are new posts from ${authorUsername}`;
 
+    const getstreamClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
+    let unreadCount = 0;
+    try {
+      await getstreamClient.connectUser({id: userId}, token);
+
+      const userChannel = getstreamClient.channel('topics', `topic_${name}`);
+      const createdChannel = await userChannel.watch();
+      const readUsers = createdChannel?.read;
+      const selfUserReadStatus = readUsers?.find((readStatus) => {
+        return readStatus?.user?.id === userId;
+      });
+
+      unreadCount = selfUserReadStatus?.unread_messages || 0;
+    } catch (e) {
+      console.error('Error getting unread count', e);
+    } finally {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error logging here is quite generic and does not provide much context about what operation failed. Consider providing more specific error messages to help with debugging.

```diff
-      console.error('Error getting unread count', e);
+      console.error(`Error getting unread count for user ${userId} on topic ${name}`, e);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->